app/hooks/navigation_button_pressed.tsx

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {DependencyList, EffectCallback, useEffect} from 'react';
+import {Navigation} from 'react-native-navigation';
+
+const useNavButtonPressed = (navButtonId: string, componentId: string, callback: EffectCallback, deps?: DependencyList) => {
+    useEffect(() => {
+        const unsubscribe = Navigation.events().registerComponentListener({
```

Comment:
The only problem I see with this is that we will be doing one different listener for each button. Probably not a big deal though, and worth the improve on readability..